<a href="https://colab.research.google.com/github/payasvi/dl/blob/master/prac8/Keras_metrics_DL_Prac_8(i).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Usage of metrics

A metric is a function that is used to judge the performance of your model. Metric functions are to be supplied in the metrics parameter when a model is compiled.

```
model.compile(loss='mean_squared_error',
              optimizer='sgd',
              metrics=['mae', 'acc'])
from keras import metrics

model.compile(loss='mean_squared_error',
              optimizer='sgd',
              metrics=[metrics.mae, metrics.categorical_accuracy])
```

A metric function is similar to a loss function, except that the results from evaluating a metric are not used when training the model. You may use any of the loss functions as a metric function.

You can either pass the name of an existing metric, or pass a Theano/TensorFlow symbolic function (see Custom metrics).

##Arguments
1. y_true: True labels. Theano/TensorFlow tensor.
2. y_pred: Predictions. Theano/TensorFlow tensor of the same shape as y_true.

##Returns

1. Single tensor value representing the mean of the output array across all datapoints.

#Accuracy
keras.metrics.accuracy(y_true, y_pred)

In [0]:
import keras
from keras import models
from keras.layers import Dense, Dropout
from keras.utils import to_categorical
from keras.datasets import mnist
from keras.utils.vis_utils import model_to_dot
from IPython.display import SVG

NUM_ROWS = 28
NUM_COLS = 28
NUM_CLASSES = 10
BATCH_SIZE = 128
EPOCHS = 10

# Load data
(X_train, y_train), (X_test, y_test) = mnist.load_data()

# Reshape data
X_train = X_train.reshape((X_train.shape[0], NUM_ROWS * NUM_COLS))
X_train = X_train.astype('float32') / 255
X_test = X_test.reshape((X_test.shape[0], NUM_ROWS * NUM_COLS))
X_test = X_test.astype('float32') / 255

# Categorically encode labels
y_train = to_categorical(y_train, NUM_CLASSES)
y_test = to_categorical(y_test, NUM_CLASSES)


# Build neural network
model = models.Sequential()
model.add(Dense(512, activation='elu', input_shape=(NUM_ROWS * NUM_COLS,)))
model.add(Dense(256, activation='elu'))
model.add(Dense(10, activation='softmax'))

# Compile model using accuracy
model.compile(optimizer='rmsprop',
              loss='categorical_crossentropy',
              metrics=['accuracy'])

# Train model
model.fit(X_train, y_train,
          batch_size=BATCH_SIZE,
          epochs=EPOCHS,
          verbose=1,
          validation_data=(X_test, y_test))

score = model.evaluate(X_test, y_test, verbose=0)
print('Test loss:', score[0])
print('Test accuracy:', score[1])

Using TensorFlow backend.


11493376/11490434 [==============================] - 1s 0us/step





Instructions for updating:
Use tf.where in 2.0, which has the same broadcast rule as np.where



Train on 60000 samples, validate on 10000 samples
Epoch 1/10





60000/60000 [==============================] - 11s 180us/step - loss: 0.2898 - acc: 0.9129 - val_loss: 0.1516 - val_acc: 0.9540
Epoch 2/10
60000/60000 [==============================] - 2s 28us/step - loss: 0.1198 - acc: 0.9633 - val_loss: 0.1325 - val_acc: 0.9583
Epoch 3/10
60000/60000 [==============================] - 2s 26us/step - loss: 0.0826 - acc: 0.9741 - val_loss: 0.1001 - val_acc: 0.9692
Epoch 4/10
60000/60000 [==============================] - 2s 25us/step - loss: 0.0629 - acc: 0.9802 - val_loss: 0.0854 - val_acc: 0.9760
Epoch 5/10
60000/60000 [==============================] - 2s 25us/step - loss: 0.0502 - acc: 0.9839 - val_loss: 0.0795 - val_acc: 0.9757
Epoch 6/10
60000/60000 [==============================] - 2s 27us/step - loss: 0.0402 - acc

#binary_accuracy

keras.metrics.binary_accuracy(y_true, y_pred, threshold=0.5)

In [0]:
# Compile model using binary_accuracy
model.compile(optimizer='rmsprop',
              loss='categorical_crossentropy',
              metrics=['binary_accuracy'])

# Train model
model.fit(X_train, y_train,
          batch_size=BATCH_SIZE,
          epochs=EPOCHS,
          verbose=1,
          validation_data=(X_test, y_test))

score = model.evaluate(X_test, y_test, verbose=0)
print('Test loss:', score[0])
print('Test accuracy:', score[1])

Train on 60000 samples, validate on 10000 samples
Epoch 1/10
60000/60000 [==============================] - 2s 29us/step - loss: 0.0196 - binary_accuracy: 0.9987 - val_loss: 0.0967 - val_binary_accuracy: 0.9959
Epoch 2/10
60000/60000 [==============================] - 2s 25us/step - loss: 0.0145 - binary_accuracy: 0.9991 - val_loss: 0.1096 - val_binary_accuracy: 0.9957
Epoch 3/10
60000/60000 [==============================] - 2s 29us/step - loss: 0.0131 - binary_accuracy: 0.9991 - val_loss: 0.1048 - val_binary_accuracy: 0.9960
Epoch 4/10
60000/60000 [==============================] - 2s 26us/step - loss: 0.0119 - binary_accuracy: 0.9992 - val_loss: 0.0988 - val_binary_accuracy: 0.9962
Epoch 5/10
60000/60000 [==============================] - 2s 26us/step - loss: 0.0109 - binary_accuracy: 0.9993 - val_loss: 0.0973 - val_binary_accuracy: 0.9962
Epoch 6/10
60000/60000 [==============================] - 2s 29us/step - loss: 0.0091 - binary_accuracy: 0.9994 - val_loss: 0.1026 - val_binary_a

#categorical_accuracy

keras.metrics.categorical_accuracy(y_true, y_pred)

In [0]:
# Compile model using categorical_accuracy
model.compile(optimizer='rmsprop',
              loss='categorical_crossentropy',
              metrics=['categorical_accuracy'])

# Train model
model.fit(X_train, y_train,
          batch_size=BATCH_SIZE,
          epochs=EPOCHS,
          verbose=1,
          validation_data=(X_test, y_test))

score = model.evaluate(X_test, y_test, verbose=0)
print('Test loss:', score[0])
print('Test accuracy:', score[1])

Train on 60000 samples, validate on 10000 samples
Epoch 1/10
60000/60000 [==============================] - 2s 28us/step - loss: 0.0068 - categorical_accuracy: 0.9979 - val_loss: 0.1126 - val_categorical_accuracy: 0.9820
Epoch 2/10
60000/60000 [==============================] - 2s 25us/step - loss: 0.0066 - categorical_accuracy: 0.9979 - val_loss: 0.1380 - val_categorical_accuracy: 0.9790
Epoch 3/10
60000/60000 [==============================] - 1s 25us/step - loss: 0.0056 - categorical_accuracy: 0.9982 - val_loss: 0.1105 - val_categorical_accuracy: 0.9828
Epoch 4/10
60000/60000 [==============================] - 2s 26us/step - loss: 0.0060 - categorical_accuracy: 0.9984 - val_loss: 0.1232 - val_categorical_accuracy: 0.9817
Epoch 5/10
60000/60000 [==============================] - 2s 26us/step - loss: 0.0059 - categorical_accuracy: 0.9982 - val_loss: 0.1277 - val_categorical_accuracy: 0.9812
Epoch 6/10
60000/60000 [==============================] - 2s 30us/step - loss: 0.0068 - categor

#top_k_categorical_accuracy

keras.metrics.top_k_categorical_accuracy(y_true, y_pred, k=5)

In [0]:
# Compile model using categorical_accuracy
model.compile(optimizer='rmsprop',
              loss='categorical_crossentropy',
              metrics=['top_k_categorical_accuracy'])

# Train model
model.fit(X_train, y_train,
          batch_size=BATCH_SIZE,
          epochs=EPOCHS,
          verbose=1,
          validation_data=(X_test, y_test))

score = model.evaluate(X_test, y_test, verbose=0)
print('Test loss:', score[0])
print('Test accuracy:', score[1])

Train on 60000 samples, validate on 10000 samples
Epoch 1/10
60000/60000 [==============================] - 2s 30us/step - loss: 0.0045 - top_k_categorical_accuracy: 1.0000 - val_loss: 0.1342 - val_top_k_categorical_accuracy: 0.9998
Epoch 2/10
60000/60000 [==============================] - 1s 25us/step - loss: 0.0036 - top_k_categorical_accuracy: 1.0000 - val_loss: 0.1332 - val_top_k_categorical_accuracy: 0.9997
Epoch 3/10
60000/60000 [==============================] - 1s 25us/step - loss: 0.0037 - top_k_categorical_accuracy: 1.0000 - val_loss: 0.1443 - val_top_k_categorical_accuracy: 1.0000
Epoch 4/10
60000/60000 [==============================] - 2s 25us/step - loss: 0.0045 - top_k_categorical_accuracy: 1.0000 - val_loss: 0.1574 - val_top_k_categorical_accuracy: 0.9998
Epoch 5/10
60000/60000 [==============================] - 2s 26us/step - loss: 0.0047 - top_k_categorical_accuracy: 1.0000 - val_loss: 0.1556 - val_top_k_categorical_accuracy: 0.9999
Epoch 6/10
60000/60000 [===========

#sparse_categorical_accuracy

keras.metrics.sparse_categorical_accuracy(y_true, y_pred)

In [0]:
from numpy import array
from keras.models import Sequential
from keras.layers import Dense
from matplotlib import pyplot
# prepare sequence
X = array([0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0])
y = array([0, 0, 0, 0, 0, 1, 1, 1, 1, 1])
# create model
model = Sequential()
model.add(Dense(2, input_dim=1))
model.add(Dense(1, activation='sigmoid'))
model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['sparse_categorical_accuracy'])
# train model
history = model.fit(X, y, epochs=400, batch_size=len(X), verbose=2)

Epoch 1/400
 - 0s - loss: 0.7885 - sparse_categorical_accuracy: 0.5000
Epoch 2/400
 - 0s - loss: 0.7879 - sparse_categorical_accuracy: 0.5000
Epoch 3/400
 - 0s - loss: 0.7874 - sparse_categorical_accuracy: 0.5000
Epoch 4/400
 - 0s - loss: 0.7868 - sparse_categorical_accuracy: 0.5000
Epoch 5/400
 - 0s - loss: 0.7862 - sparse_categorical_accuracy: 0.5000
Epoch 6/400
 - 0s - loss: 0.7856 - sparse_categorical_accuracy: 0.5000
Epoch 7/400
 - 0s - loss: 0.7851 - sparse_categorical_accuracy: 0.5000
Epoch 8/400
 - 0s - loss: 0.7845 - sparse_categorical_accuracy: 0.5000
Epoch 9/400
 - 0s - loss: 0.7840 - sparse_categorical_accuracy: 0.5000
Epoch 10/400
 - 0s - loss: 0.7834 - sparse_categorical_accuracy: 0.5000
Epoch 11/400
 - 0s - loss: 0.7829 - sparse_categorical_accuracy: 0.5000
Epoch 12/400
 - 0s - loss: 0.7823 - sparse_categorical_accuracy: 0.5000
Epoch 13/400
 - 0s - loss: 0.7818 - sparse_categorical_accuracy: 0.5000
Epoch 14/400
 - 0s - loss: 0.7813 - sparse_categorical_accuracy: 0.5000
E

#sparse_top_k_categorical_accuracy

keras.metrics.sparse_top_k_categorical_accuracy(y_true, y_pred, k=5)

In [0]:
from numpy import array
from keras.models import Sequential
from keras.layers import Dense
from matplotlib import pyplot
# prepare sequence
X = array([0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0])
y = array([0, 0, 0, 0, 0, 1, 1, 1, 1, 1])
# create model
model = Sequential()
model.add(Dense(2, input_dim=1))
model.add(Dense(1, activation='sigmoid'))
model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['sparse_top_k_categorical_accuracy'])
# train model
history = model.fit(X, y, epochs=400, batch_size=len(X), verbose=2)

Epoch 1/400
 - 0s - loss: 0.7121 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 2/400
 - 0s - loss: 0.7118 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 3/400
 - 0s - loss: 0.7115 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 4/400
 - 0s - loss: 0.7111 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 5/400
 - 0s - loss: 0.7107 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 6/400
 - 0s - loss: 0.7103 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 7/400
 - 0s - loss: 0.7099 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 8/400
 - 0s - loss: 0.7096 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 9/400
 - 0s - loss: 0.7092 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 10/400
 - 0s - loss: 0.7088 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 11/400
 - 0s - loss: 0.7084 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 12/400
 - 0s - loss: 0.7080 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 13/400
 - 0s - loss: 0.7076 - sparse_top_k_categorical_accuracy: 0.

#cosine_proximity

keras.metrics.cosine_proximity(y_true, y_pred, axis=-1)

In [0]:
from numpy import array
from keras.models import Sequential
from keras.layers import Dense
from matplotlib import pyplot
# prepare sequence
X = array([0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0])
y = array([0, 0, 0, 0, 0, 1, 1, 1, 1, 1])
# create model
model = Sequential()
model.add(Dense(2, input_dim=1))
model.add(Dense(1, activation='sigmoid'))
model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['cosine_proximity'])
# train model
history = model.fit(X, y, epochs=400, batch_size=len(X), verbose=2)

Epoch 1/400
 - 0s - loss: 0.7432 - cosine_proximity: -5.0000e-01
Epoch 2/400
 - 0s - loss: 0.7429 - cosine_proximity: -5.0000e-01
Epoch 3/400
 - 0s - loss: 0.7425 - cosine_proximity: -5.0000e-01
Epoch 4/400
 - 0s - loss: 0.7421 - cosine_proximity: -5.0000e-01
Epoch 5/400
 - 0s - loss: 0.7417 - cosine_proximity: -5.0000e-01
Epoch 6/400
 - 0s - loss: 0.7413 - cosine_proximity: -5.0000e-01
Epoch 7/400
 - 0s - loss: 0.7409 - cosine_proximity: -5.0000e-01
Epoch 8/400
 - 0s - loss: 0.7405 - cosine_proximity: -5.0000e-01
Epoch 9/400
 - 0s - loss: 0.7401 - cosine_proximity: -5.0000e-01
Epoch 10/400
 - 0s - loss: 0.7397 - cosine_proximity: -5.0000e-01
Epoch 11/400
 - 0s - loss: 0.7393 - cosine_proximity: -5.0000e-01
Epoch 12/400
 - 0s - loss: 0.7389 - cosine_proximity: -5.0000e-01
Epoch 13/400
 - 0s - loss: 0.7385 - cosine_proximity: -5.0000e-01
Epoch 14/400
 - 0s - loss: 0.7381 - cosine_proximity: -5.0000e-01
Epoch 15/400
 - 0s - loss: 0.7377 - cosine_proximity: -5.0000e-01
Epoch 16/400
 - 0s 

#Thank you for completing the notebook

*ACCURACY SCORES*

**Accuracy-**

Test loss: 0.10376196658944027


Test accuracy: 0.9757


**binary_accuracy-**

Test loss: 0.15646206490577766


Test accuracy: 0.9955399974822998


**categorical_accuracy-**

Test loss: 0.14195446051103405


Test accuracy: 0.9816


**top_k_categorical_accuracy-**

Test loss: 0.15224556388646485


Test accuracy: 0.9999


**sparse_categorical_accuracy-**

loss: 0.4375 


sparse_categorical_accuracy: 0.5000


**sparse_top_k_categorical_accuracy-**

loss: 0.7547 


sparse_top_k_categorical_accuracy: 0.5000


**cosine_proximity-**

loss: 0.7574 


cosine_proximity: -5.0000e-01

The best metric among all is **top_k_categorical_accuracy** which gives almost an accuracy of 1.

In [0]:
from keras.datasets import cifar10
from keras.models import Sequential
from keras.layers import Dense, Dropout, Activation, Flatten
from keras.utils import to_categorical

#import dataset
(X_train, Y_train), (X_test, Y_test) = cifar10.load_data()

#change shape from image to vector
X_train = X_train.reshape(50000, 32 * 32 * 3)
X_test = X_test.reshape(10000, 32 * 32 * 3)

#preprocess
X_train = X_train.astype('float32')
X_test = X_test.astype('float32')
X_train /= 255.0
X_test /= 255.0

#change labels from numeric to one hot encoded
Y_train = to_categorical(Y_train, 10)
Y_test =  to_categorical(Y_test, 10)

#Model building
model = Sequential()
model.add(Dense(1024, input_shape=(3072, )))
model.add(Activation('relu'))
model.add(Dense(512))
model.add(Activation('relu'))
model.add(Dense(512))
model.add(Activation('relu'))
model.add(Dense(10))
model.add(Activation('softmax'))

# Compile model using above optimizer
model.compile(optimizer='rmsprop',
              loss='categorical_crossentropy',
              metrics=['top_k_categorical_accuracy'])

# Train model
model.fit(X_train, Y_train,
          batch_size=BATCH_SIZE,
          epochs=EPOCHS,
          verbose=1,
          validation_data=(X_test, Y_test))

score = model.evaluate(X_test, Y_test, verbose=0)
print('Test loss:', score[0])
print('Test accuracy:', score[1])

170500096/170498071 [==============================] - 4s 0us/step
Train on 50000 samples, validate on 10000 samples
Epoch 1/10
50000/50000 [==============================] - 2s 50us/step - loss: 2.4006 - top_k_categorical_accuracy: 0.7490 - val_loss: 1.8313 - val_top_k_categorical_accuracy: 0.8438
Epoch 2/10
50000/50000 [==============================] - 2s 41us/step - loss: 1.8006 - top_k_categorical_accuracy: 0.8496 - val_loss: 1.7774 - val_top_k_categorical_accuracy: 0.8473
Epoch 3/10
50000/50000 [==============================] - 2s 43us/step - loss: 1.6997 - top_k_categorical_accuracy: 0.8728 - val_loss: 1.6803 - val_top_k_categorical_accuracy: 0.8772
Epoch 4/10
50000/50000 [==============================] - 2s 41us/step - loss: 1.6284 - top_k_categorical_accuracy: 0.8870 - val_loss: 1.5789 - val_top_k_categorical_accuracy: 0.8959
Epoch 5/10
50000/50000 [==============================] - 2s 42us/step - loss: 1.5710 - top_k_categorical_accuracy: 0.8996 - val_loss: 1.6857 - val_top

**CIFAR 10**

Test loss: 1.4908046955108643


Test accuracy: 0.914

In [0]:
from keras.datasets import cifar100
from keras.models import Sequential
from keras.layers import Dense, Dropout, Activation, Flatten
from keras.utils import to_categorical

#import dataset
(X_train, Y_train), (X_test, Y_test) = cifar100.load_data()

#change shape from image to vector
X_train = X_train.reshape(50000, 32 * 32 * 3)
X_test = X_test.reshape(10000, 32 * 32 * 3)

#preprocess
X_train = X_train.astype('float32')
X_test = X_test.astype('float32')
X_train /= 255.0
X_test /= 255.0

#change labels from numeric to one hot encoded
Y_train = to_categorical(Y_train, 100)
Y_test =  to_categorical(Y_test, 100)

#Model building
model = Sequential()
model.add(Dense(1024, input_shape=(3072, )))
model.add(Activation('relu'))
model.add(Dense(512))
model.add(Activation('relu'))
model.add(Dense(512))
model.add(Activation('relu'))
model.add(Dense(100))
model.add(Activation('softmax'))


from keras import optimizers
adamax=keras.optimizers.Adamax(lr=0.002, beta_1=0.9, beta_2=0.999)

# Compile model using above optimizer
model.compile(optimizer='rmsprop',
              loss='categorical_crossentropy',
              metrics=['top_k_categorical_accuracy'])

# Train model
model.fit(X_train, Y_train,
          batch_size=BATCH_SIZE,
          epochs=EPOCHS,
          verbose=1,
          validation_data=(X_test, Y_test))

score = model.evaluate(X_test, Y_test, verbose=0)
print('Test loss:', score[0])
print('Test accuracy:', score[1])

169009152/169001437 [==============================] - 4s 0us/step
Train on 50000 samples, validate on 10000 samples
Epoch 1/10
50000/50000 [==============================] - 3s 52us/step - loss: 4.2765 - top_k_categorical_accuracy: 0.2107 - val_loss: 3.8325 - val_top_k_categorical_accuracy: 0.3267
Epoch 2/10
50000/50000 [==============================] - 2s 41us/step - loss: 3.7775 - top_k_categorical_accuracy: 0.3411 - val_loss: 3.7126 - val_top_k_categorical_accuracy: 0.3562
Epoch 3/10
50000/50000 [==============================] - 2s 41us/step - loss: 3.5882 - top_k_categorical_accuracy: 0.3948 - val_loss: 3.7838 - val_top_k_categorical_accuracy: 0.3731
Epoch 4/10
50000/50000 [==============================] - 2s 42us/step - loss: 3.4546 - top_k_categorical_accuracy: 0.4326 - val_loss: 3.5597 - val_top_k_categorical_accuracy: 0.4101
Epoch 5/10
50000/50000 [==============================] - 2s 40us/step - loss: 3.3407 - top_k_categorical_accuracy: 0.4653 - val_loss: 3.5403 - val_top

**CIFAR 100**

Test loss: 3.393477967834473


Test accuracy: 0.483

In [0]:
import keras
from keras import models
from keras.layers import Dense, Dropout
from keras.utils import to_categorical
from keras.datasets import fashion_mnist
from keras.utils.vis_utils import model_to_dot
from IPython.display import SVG

NUM_ROWS = 28
NUM_COLS = 28
NUM_CLASSES = 10
BATCH_SIZE = 128
EPOCHS = 10

# Load data
(X_train, Y_train), (X_test, Y_test) = fashion_mnist.load_data()



# Reshape data
X_train = X_train.reshape((X_train.shape[0], NUM_ROWS * NUM_COLS))
X_train = X_train.astype('float32') / 255
X_test = X_test.reshape((X_test.shape[0], NUM_ROWS * NUM_COLS))
X_test = X_test.astype('float32') / 255

# Categorically encode labels
Y_train = to_categorical(Y_train, NUM_CLASSES)
Y_test = to_categorical(Y_test, NUM_CLASSES)


# Build neural network
model = models.Sequential()
model.add(Dense(512, activation='relu', input_shape=(NUM_ROWS * NUM_COLS,)))
model.add(Dense(256, activation='relu'))
model.add(Dense(10, activation='softmax'))

# Compile model using above optimizer
model.compile(optimizer='rmsprop',
              loss='categorical_crossentropy',
              metrics=['top_k_categorical_accuracy'])

# Train model
model.fit(X_train, Y_train,
          batch_size=BATCH_SIZE,
          epochs=EPOCHS,
          verbose=1,
          validation_data=(X_test, Y_test))

score = model.evaluate(X_test, Y_test, verbose=0)
print('Test loss:', score[0])
print('Test accuracy:', score[1])

4423680/4422102 [==============================] - 1s 0us/step
Train on 60000 samples, validate on 10000 samples
Epoch 1/10
60000/60000 [==============================] - 2s 35us/step - loss: 0.5421 - top_k_categorical_accuracy: 0.9927 - val_loss: 0.4280 - val_top_k_categorical_accuracy: 0.9968
Epoch 2/10
60000/60000 [==============================] - 1s 25us/step - loss: 0.3764 - top_k_categorical_accuracy: 0.9977 - val_loss: 0.4121 - val_top_k_categorical_accuracy: 0.9983
Epoch 3/10
60000/60000 [==============================] - 2s 26us/step - loss: 0.3355 - top_k_categorical_accuracy: 0.9983 - val_loss: 0.3812 - val_top_k_categorical_accuracy: 0.9968
Epoch 4/10
60000/60000 [==============================] - 2s 27us/step - loss: 0.3102 - top_k_categorical_accuracy: 0.9986 - val_loss: 0.3425 - val_top_k_categorical_accuracy: 0.9975
Epoch 5/10
60000/60000 [==============================] - 2s 25us/step - loss: 0.2929 - top_k_categorical_accuracy: 0.9990 - val_loss: 0.3883 - val_top_k_c

**FASHION_MNIST**

Test loss: 0.34572943717837334


Test accuracy: 0.9981